federation-js/src/service/printSupergraphSdl.ts

```diff
@@ -218,40 +237,50 @@ function printFederationTypeDirectives(type: GraphQLObjectType): string {
 
   // Separate owner @keys from the rest of the @keys so we can print them
   // adjacent to the @owner directive.
-  const { [ownerService]: ownerKeys = [], ...restKeys } = keys
-  const ownerEntry: [string, (readonly SelectionNode[])[]] = [ownerService, ownerKeys];
+  const { [ownerService]: ownerKeys = [], ...restKeys } = keys;
+  const ownerEntry: [string, (readonly SelectionNode[])[]] = [
+    ownerService,
+    ownerKeys,
+  ];
   const restEntries = Object.entries(restKeys);
 
+  // We don't want to print an owner for interface types
+  const shouldPrintOwner = isObjectType(type);
+  const joinOwnerString = shouldPrintOwner
+    ? `\n  @join__owner(graph: ${
+        context.sanitizedServiceNames?.[ownerService] ?? ownerService
```

Comment:
In what context does this `?? ownerService` trigger? It seems like this should be an error rather than creating references to a non-existent enum value.

Same for the other three similar calls, though one could imagine a helper function or method on `context` or something that does the lookup-and-error-check (or the lookup-and-fallback as done here).